webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/users/UserRepository.scala

```diff
@@ -722,6 +744,29 @@ class JdbcUserRepository(doobie: Doobie) extends UserRepository {
       trace:           EventTrace,
       isCaseSensitive: Boolean
   ): IOResult[Set[String]] = {
+    transactIOResult("Error when updating the list of users") { xa =>
+      setUsers(origin, users, trace, isCaseSensitive).transact(xa)
+    }
+  }
+
+  override def addUser(origin: String, user: String, trace: EventTrace, isCaseSensitive: Boolean): IOResult[Boolean] = {
+    transactIOResult(s"Error when adding user '${user}' from '${origin}'") { xa =>
+      (for {
+        current  <- fr"select id from users where managedby = ${origin}".query[String].to[List]
+        added     = current :+ user
+        notAdded <- setUsers(origin, added, trace, isCaseSensitive)
```

Comment:
The `setUsers` does many consistency checks on all the existing base of users, also taking the case sensitivity into account.
It may be possible but would need much more code factorization, to avoid querying the whole base of users when adding every user.
So, as with the `InMemory` repo, we would need to reason on having a whole base, there are less constructs when doing a `add` that reuses the `set` method that does all the checks, than when doing a `set` that reuse the `add` + other specific constructs for the checks

Reply:
ok, fair enough